ARCHITECTURE.md

```diff
@@ -0,0 +1,137 @@
+# Architecture overview
+
+This document provides an overview of all parts and pueces of the project as well as how they fit together. It is meant to help new contributors understand where things may be, and how changes in some components may interact with other components of the system.
+
+## Root package (`sqle`)
+
+This is where the engine lives. The engine is the piece that coordinates and makes all other pieces work together as well as the main API users of the system will use to create and configure an engine and perform queries.
+
+Because this is the point where all components fit together, it is also where integration tests are. Those integration tests can be found in `engine_test.go`.
+Every time a feature is added or a bug that comes from a particular query is fixed, a test should be added there.
```

Comment:
Maybe: "a test should be added there, plus in any specific where the feature/issue belonged, if needed".